### ISSUE TITLE
Polygons and MultiPolygons should follow the right-hand rule

### ISSUE THREAD

**Issue by ckchow** [ropensci/geojsonlint#10]
The website at http://geojsonlint.com/ will complain on the below json with the error that polygons should follow the right hand rule.

`
{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-84.39165115356445,33.78150156321427],[-84.39049243927,33.77122807629594],[-84.39602851867676,33.77126375025704],[-84.40362453460692,33.77433165534681],[-84.40722942352295,33.77764915000493],[-84.40735816955566,33.78143022380356],[-84.39165115356445,33.78150156321427]]]}}
`
According to the new spec  https://tools.ietf.org/html/rfc7946 (section 3.1.6: Polygon)

>    Note: the [GJ2008] specification did not discuss linear ring winding
   order.  For backwards compatibility, parsers SHOULD NOT reject
   Polygons that do not follow the right-hand rule.

Maybe the linter should warn for standards-compliance reasons but still draw the bad polygon.

**Comment by sckott** [comment]
Thanks @ckchow for the issue

We have three methods in this pkg

* `geojsonhint` JS library, `geojsonlint::geojson_hint` - This method follows RFC 7946 and does throw warning about the right hand rule, e.g., try 

```r
geojson_hint('{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-84.39165115356445,33.78150156321427],[-84.39049243927,33.77122807629594],[-84.39602851867676,33.77126375025704],[-84.40362453460692,33.77433165534681],[-84.40722942352295,33.77764915000493],[-84.40735816955566,33.78143022380356],[-84.39165115356445,33.78150156321427]]]}}', verbose = TRUE)
#> [1] FALSE
#> attr(,"errors")
#>   line   level                                                      message
#> 1    1 message Polygons and MultiPolygons should follow the right-hand rule
```

> note, just updated to v2 of `geojsonhint` JS library, was on v2beta before, so reinstall from github `devtools::install_github("ropenscilabs/geojsonlint")`

* `is-my-json-valid` JS library, `geojsonlint::geojson_validate` - which doesn't catch right hand rule right now
* geojsonlint API, `geojsonlint::geojson_lint` (which I thought was the same as the http://geojsonlint.com/ interface, but I don't get an error with your example geojson above when using the API. 

```
➜  ~ curl -XPOST 'http://geojsonlint.com/validate' -d  '{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-84.39165115356445,33.78150156321427],[-84.39049243927,33.77122807629594],[-84.39602851867676,33.77126375025704],[-84.40362453460692,33.77433165534681],[-84.40722942352295,33.77764915000493],[-84.40735816955566,33.78143022380356],[-84.39165115356445,33.78150156321427]]]}}' | jq .
{
  "status": "ok"
}
```

not sure why the behavior is different.

related: working on port of mapbox's `geojson-rewind` for R https://github.com/ropenscilabs/geojsonrewind

**Comment by answerquest** [comment]
For folks who got here looking for a fix, this site has a client-side JS that reverses the co-ordinates where needed : https://mapster.me/right-hand-rule-geojson-fixer/

**Comment by matkoniecz** [comment]
> The website at http://geojsonlint.com/ will complain on the below json with the error that polygons should follow the right hand rule.

> Maybe the linter should warn for standards-compliance reasons but still draw the bad polygon.

I am confused a bit - https://github.com/ropensci/geojsonlint has "(THIS IS NOT THE SRC FOR geojsonlint.com)", so what this site is doing, should be offtopic here, right?

**Comment by sckott** [comment]
yes, i don't remember why i responded the way I did, in general yes, the geojsonlint site is off topic here, but if there's a question of different reusults between it and this pkg, that might be of interest
